server/vulnerabilities/oval/parsed/rpmvercmp_test.go

```diff
@@ -0,0 +1,4564 @@
+package oval_parsed
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestVersionParts(t *testing.T) {
+	cases := []struct {
+		v       string
+		epoch   int
+		version string
+		release string
+	}{
+		{"0.0", 0, "0.0", ""},
+		{"0", 0, "0", ""},
+		{"0:0", 0, "0", ""},
+		{"0:0-", 0, "0", ""},
+		{"0:0-0", 0, "0", "0"},
+		{"0:0.0-0.0", 0, "0.0", "0.0"},
+		{"1:0", 1, "0", ""},
+		{"5:1", 5, "1", ""},
+		{"0:0-0-0", 0, "0", "0-0"},
+		{"0:0-0-0-0", 0, "0", "0-0-0"},
+		{"0:0:0-0", 0, "0:0", "0"},
+		{"0:0:0:0-0", 0, "0:0:0", "0"},
+		{"0:0:0-0-0", 0, "0:0", "0-0"},
+		{"0:0-0:0-0", 0, "0", "0:0-0"},
+		{"  	0:0-1", 0, "0", "1"},
+		{"0:0-1  	", 0, "0", "1"},
+		{"	  0:0-1  	", 0, "0", "1"},
+		{"", 0, "", ""},
+		{" ", 0, "", ""},
+		{"0:", 0, "", ""},
+		{"0:0 0-1", 0, "0 0", "1"},
+		{"-1:0-1", -1, "0", "1"},
+		{"a:0-0", 0, "0", "0"},
+		{"A:1-2", 0, "1", "2"},
+		{"0:abc3-0", 0, "abc3", "0"},
+		{"1.2.3", 0, "1.2.3", ""},
+		{"1:1.2.3", 1, "1.2.3", ""},
+		{"A:1.2.3", 0, "1.2.3", ""},
+		{"-1:1.2.3", -1, "1.2.3", ""},
+		{"6.0-4.el6.x86_64", 0, "6.0", "4.el6.x86_64"},
+		{"c105b9de-4e0fd3a3", 0, "c105b9de", "4e0fd3a3"},
+		{"4.999.9-0.5.beta.20091007git.el6", 0, "4.999.9", "0.5.beta.20091007git.el6"},
+	}
+
+	for _, c := range cases {
+		require.Equal(t, c.epoch, epoch(c.v))
+		require.Equal(t, c.version, version(c.v))
+		require.Equal(t, c.release, release(c.v))
+	}
+}
+
+func TestRpmvercmp(t *testing.T) {
```

Comment:
Cool!, what's the source for the test vector?

Reply:
This https://github.com/knqyf263/go-rpm-version/blob/master/version_test.go but had to update a bunch of bad test cases - as the source of truth I used https://archlinux.org/packages/community/x86_64/rpm-tools/

Reply:
Cool, let's add link/s (sorry if it's already there I missed it :)